### PR TITLE
Hook prepareForSegue with Aspects instead of RAC

### LINKED ISSUE #3: Remove dependency on ReactiveCocoa
We currently use RAC to dynamically implement `prepareForSegue(_:sender:)`, but it's not exposed via the public API. It really simplifies this library, but it would be nice to investigate removing that dependency before 1.0.

### PR DESCRIPTION
**Pull request by sharplet**
Fixes #3.

Aspects is a significantly thinner library than RAC, yet still provides the feature we're interested in: a block-based callback for a message-send.  This even simplifies the implementation by discarding the extra power (and thus complexity) that RAC provides.

### CODE REVIEW

File: Sources/Perform/Aspects.swift [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import Aspects
+
+internal extension NSObject {
```
Comment (jakecraige) [inline Sources/Perform/Aspects.swift]: Do we need to be explicit about `internal` or is that the default?

Comment (sharplet) [reply Sources/Perform/Aspects.swift]: It's the default, I suppose there's no real need to be explicit here

**Comment by sharplet** [comment]
For simplicity's sake, it's worth considering whether we should go the extra mile and make Aspects a completely private dependency.

**Comment by sharplet** [comment]
> For simplicity's sake, it's worth considering whether we should go the extra mile and make Aspects a completely private dependency.

Actually, to be honest, I don't think this is worth the extra maintenance burden.

**Comment by jakecraige** [comment]
👍 , cool find of this framework

**Pull request merged**